recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -142,6 +142,16 @@ this.PreferenceExperiments = {
       previousPreferenceValue: Preferences.get(preferenceName, undefined),
     };
 
+    if (experiment.previousPreferenceValue) {
+      const previousType = typeof experiment.previousPreferenceValue;
+      const givenType = typeof preferenceValue;
```

Comment:
Alright, I've corrected the approach and use the strings we use in the recipe arguments accordingly. _However,_ I'm having trouble figuring out how to actually read the recipe's arguments from within `PreferenceExperiments.jsm`. I've found that the RecipeRunner has access to the `recipe` object, but I couldn't figure out how to pass that into the PrefExperiments file.

Any suggestions?